R/mlflow/R/model.R

```diff
@@ -0,0 +1,167 @@
+#' Save Model for MLflow
+#'
+#' Saves model in MLflow's format that can later be used
+#' for prediction and serving.
+#'
+#' @param x The serving function or model that will perform a prediction.
+#' @param path Destination path where this MLflow compatible model
+#'   will be saved.
+#' @param dependencies Optional vector of paths to dependency files
+#'   to include in the model, as in \code{r-dependencies.txt}
+#'   or \code{conda.yaml}.
+#'
+#' @importFrom yaml write_yaml
+#' @export
+mlflow_save_model <- function(x, path = "model", dependencies = NULL) {
+  supported_deps <- list(
+    "r-dependencies.txt" = "r_dependencies",
+    "conda.yaml" = "conda_env"
+  )
+
+  if (!is.null(dependencies)) {
+    dep_file <- basename(dependencies)
+    if (!all(dep_file %in% names(supported_deps))) {
+      stop("Dependency ", dep_file, " is unsupported. Supported files: ", paste(names(supported_deps), collapse = ", "))
+    }
+  }
+
+  if (dir.exists(path)) unlink(path, recursive = TRUE)
+  dir.create(path)
+
+  flavor_spec <- list (
+    flavors = mlflow_save_flavor(x, path)
+  )
+
+  for (dependency in dependencies) {
+    flavor_spec[[supported_deps[[basename(dependency)]]]] <- dependency
+  }
+
+  mlflow_write_model_spec(path, flavor_spec)
+}
+
+mlflow_write_model_spec <- function(path, content) {
```

Comment:
The way we call this is as `mlflow_write_model_spec(path, flavor_spec)` and the content is the dynamic list of flavors, so I believe we are good. In any case, is marked as `nit` since this is a private API so we should be fine to leave this ASIS.